### PR TITLE
Config parser

### PR DESCRIPTION
**Pull request by vpodzime**
Based on the internal JSON API work, only the last commit is relevant here.

### CODE REVIEW

**Comment by kacf** [comment]
The direction looks good to me 👍 (only checked last commit, as instructed)!

**Comment by oleorhagen** [comment]
Looks like a good start :)

**Comment by oleorhagen** [comment]
@kacf there are a few minor TODO's left, other than that I think it is starting to look real good. I will assign myself and just push this across the line.

File: common/config_parser.hpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	string boot_utilities_set_active_part;
+
+	/** Command to get the partition which will boot next */
+	string boot_utilities_get_next_active_part;
+
+	/** Path to the device type file */
+	string device_type_file;
+
+	/** DBus configuration */
+	// TODO: struct DbusConfig { bool enabled; }; ?
```
Comment (oleorhagen) [inline common/config_parser.hpp:91]: I believe this is fine to keep as is as well.

**Review by oleorhagen — COMMENTED** [review]
I think it looks real good.

I will take it over from here.

**Comment by oleorhagen** [comment]
Closing in favour of me taking it over here: https://github.com/mendersoftware/mender/pull/1124

**Pull request closed without merging**